retrofit/src/test/java/retrofit/CallTest.java

```diff
@@ -445,18 +443,18 @@
         .build();
     Service example = retrofit.create(Service.class);
 
-    server.enqueue(new MockResponse().setBody("1234").throttleBody(1, 500, MILLISECONDS));
+    server.enqueue(new MockResponse()
+        .setBody("1234")
+        .setSocketPolicy(DISCONNECT_DURING_RESPONSE_BODY));
 
-    long exeuteStart = System.nanoTime();
     Response<ResponseBody> response = example.getStreamingBody().execute();
-    long executeTook = System.nanoTime() - exeuteStart;
-    assertThat(executeTook).isLessThan(MILLISECONDS.toNanos(500));
-
-    long readStart = System.nanoTime();
-    String body = response.body().string();
-    long readTook = System.nanoTime() - readStart;
-    assertThat(readTook).isGreaterThan(MILLISECONDS.toNanos(1000));
-    assertThat(body).isEqualTo("1234");
+
+    // When streaming we only detect socket problems as the ResponseBody is read.
+    try {
+      response.body().string();
```

Comment:
At one point I checked out the IntelliJ source code with the hope of making `catch (Exception expected)` with the name _expected_ trigger some static analysis. Maybe I should just file a feature request.